x-pack/plugins/infra/server/saved_objects/metrics_explorer_view/types.ts

```diff
@@ -8,17 +8,65 @@
 import { isoToEpochRt, nonEmptyStringRt } from '@kbn/io-ts-utils';
 import * as rt from 'io-ts';
 
-export const metricsExplorerViewSavedObjectAttributesRT = rt.intersection([
-  rt.strict({
+const metricsExplorerSavedObjectChartTypeRT = rt.keyof({ line: null, area: null, bar: null });
+const metricsExplorerYAxisModeRT = rt.keyof({ fromZero: null, auto: null });
+
+const metricsExplorerSavedObjectChartOptionsRT = rt.type({
+  yAxisMode: metricsExplorerYAxisModeRT,
+  type: metricsExplorerSavedObjectChartTypeRT,
+  stack: rt.boolean,
+});
+
+export const metricsExplorerSavedObjectTimeOptionsRT = rt.type({
+  from: rt.string,
+  to: rt.string,
+  interval: rt.string,
+});
+const metricsExplorerSavedObjectOptionsMetricRT = rt.intersection([
+  rt.UnknownRecord,
+  rt.partial({
+    rate: rt.boolean,
+    color: rt.string,
+    label: rt.string,
+  }),
+]);
+
+const metricExplorerSavedObjectOptionsRequiredRT = rt.type({
+  aggregation: rt.string,
+  metrics: rt.array(metricsExplorerSavedObjectOptionsMetricRT),
+});
+
+const metricExplorerSavedObjectOptionsOptionalRT = rt.partial({
+  limit: rt.number,
+  groupBy: rt.union([rt.string, rt.array(rt.string)]),
+  filterQuery: rt.string,
+  source: rt.string,
+  forceInterval: rt.boolean,
+  dropLastBucket: rt.boolean,
+});
+export const metricsExplorerSavedObjectOptionsRT = rt.intersection([
+  metricExplorerSavedObjectOptionsRequiredRT,
+  metricExplorerSavedObjectOptionsOptionalRT,
+]);
+
+const metricExplorerViewsSavedObjectStateRT = rt.type({
+  chartOptions: metricsExplorerSavedObjectChartOptionsRT,
+  currentTimerange: metricsExplorerSavedObjectTimeOptionsRT,
+  options: metricsExplorerSavedObjectOptionsRT,
+});
+
+const metricsExplorerViewASavedObjectAtttributesRT = rt.intersection([
+  metricExplorerViewsSavedObjectStateRT,
+  rt.type({
     name: nonEmptyStringRt,
   }),
-  rt.UnknownRecord,
+  rt.partial({ isDefault: rt.boolean, isStatic: rt.boolean }),
 ]);
 
 export const metricsExplorerViewSavedObjectRT = rt.intersection([
   rt.type({
     id: rt.string,
-    attributes: metricsExplorerViewSavedObjectAttributesRT,
+    attributes: metricsExplorerViewASavedObjectAtttributesRT,
```

Comment:
```suggestion
    attributes: metricsExplorerViewSavedObjectAttributesRT,
```

one too many T's ;).  Also there is an "A" between View and Saved.  Not sure if that's a typo or intentional?

Reply:
typo! :) thanks